lib/Serialization/ModuleDependencyScanner.cpp

```diff
@@ -176,11 +176,12 @@ Optional<ModuleDependencies> SerializedModuleLoaderBase::getModuleDependencies(
   auto moduleId = Ctx.getIdentifier(moduleName);
   // Instantiate dependency scanning "loaders".
   SmallVector<std::unique_ptr<ModuleDependencyScanner>, 2> scanners;
-  scanners.push_back(std::make_unique<ModuleDependencyScanner>(
-      Ctx, LoadMode, moduleId, delegate));
   scanners.push_back(std::make_unique<PlaceholderSwiftModuleScanner>(
       Ctx, LoadMode, moduleId, Ctx.SearchPathOpts.PlaceholderDependencyModuleMap,
       delegate));
+  scanners.push_back(std::make_unique<ModuleDependencyScanner>(
+      Ctx, LoadMode, moduleId, delegate));
+
```

Comment:
Could you add some comments about the rational? And add an asserts that the first scanner is always `ModuleDependencyScanner`.

Reply:
I think you meant the first scanner is always `PlaceholderSwiftModuleScanner `. :)
Done.

Reply:
yeah, Thank you!